src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserManagementService.cs

```diff
@@ -239,5 +317,37 @@ private void SendInviteEmail(string email)
 
             _emailService.SendEmail(email, template, emailValues);
         }
+
+        private void SendNewReleaseRoleEmail(Guid userId, IdTitlePair publication, IdTitlePair release, EnumExtensions.EnumValue role)
+        {
+            var uri = _configuration.GetValue<string>("AdminUri");
+            var template = _configuration.GetValue<string>("NotifyReleaseRoleTemplateId");
+            var email = _usersAndRolesDbContext.Users.FirstOrDefault(x => x.Id == userId.ToString())?.Email;
+
+            var emailValues = new Dictionary<string, dynamic>
+            {
+                {"url", "https://" + uri + "/publication/" + publication.Id + "/release/" + release.Id + "/summary"},
+                {"role", role.Name},
+                {"publication", publication.Title},
+                {"release", release.Title}
+            };
+
+            _emailService.SendEmail(email, template, emailValues);
+        }
+
+        private async Task<Either<ActionResult, bool>> CheckIfUserAlreadyHasReleaseRole(Guid userId,
+            UserReleaseRoleSubmission userReleaseRole)
+        {
+            var existing = await _contentDbContext.UserReleaseRoles.FirstOrDefaultAsync(r =>
+                r.UserId == userId && r.ReleaseId == userReleaseRole.ReleaseId &&
+                r.Role == userReleaseRole.ReleaseRole);
+
+            if (existing == null)
```

Comment:
This doesn't look right. The method is called `CheckIfUserAlreadyHasReleaseRole`, so should it be `if (existing != null) { return ValidationActionResult(UserAlreadyHasReleaseRole } return false` ?

Reply:
Case of bad naming i think, the logic does work